_includes/pbc/all/install-features/202304.0/marketplace/install-the-marketplace-product-offer-feature.md

```diff
@@ -0,0 +1,1822 @@
+            new ProductOfferSynchronizationDataBulkRepositoryPlugin(),
+        ];
+    }
+}
+```
+
+Configure synchronization storage:
+
+**src/Pyz/Zed/ProductOfferStorage/ProductOfferStorageConfig.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\ProductOfferStorage;
+
+use Pyz\Zed\Synchronization\SynchronizationConfig;
+use Spryker\Shared\Publisher\PublisherConfig;
+use Spryker\Zed\ProductOfferStorage\ProductOfferStorageConfig as SprykerProductOfferStorageConfig;
+
+class ProductOfferStorageConfig extends SprykerProductOfferStorageConfig
+{
+    /**
+     * @return string|null
+     */
+    public function getMerchantProductOfferSynchronizationPoolName(): ?string
+    {
+        return SynchronizationConfig::DEFAULT_SYNCHRONIZATION_POOL_NAME;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getEventQueueName(): ?string
+    {
+        return PublisherConfig::PUBLISH_QUEUE;
+    }
+}
+```
+
+Configure synchronization search
+
+**src/Pyz/Zed/MerchantProductOfferSearch/MerchantProductOfferSearchConfig.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\MerchantProductOfferSearch;
+
+use Spryker\Shared\Publisher\PublisherConfig;
+use Spryker\Zed\MerchantProductOfferSearch\MerchantProductOfferSearchConfig as SprykerMerchantProductOfferSearchConfig;
+
+class MerchantProductOfferSearchConfig extends SprykerMerchantProductOfferSearchConfig
+{
+    /**
+     * @return string|null
+     */
+    public function getMerchantEventQueueName(): ?string
+    {
+        return PublisherConfig::PUBLISH_QUEUE;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getMerchantProductOfferEventQueueName(): ?string
+    {
+        return PublisherConfig::PUBLISH_QUEUE;
+    }
+}
+```
+
+Product feature setup 
+
+**src/Pyz/Zed/Product/ProductDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\Product;
+
+use Spryker\Zed\MerchantProductOffer\Communication\Plugin\Product\MerchantProductOfferProductConcreteExpanderPlugin;
+use Spryker\Zed\Product\ProductDependencyProvider as SprykerProductDependencyProvider;
+
+class ProductDependencyProvider extends SprykerProductDependencyProvider
+{
+    /**
+     * @return array<\Spryker\Zed\ProductExtension\Dependency\Plugin\ProductConcreteExpanderPluginInterface>
+     */
+    protected function getProductConcreteExpanderPlugins(): array
+    {
+        return [
+            new MerchantProductOfferProductConcreteExpanderPlugin(),
+        ];
+    }
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure that after setting up the event listeners, the following commands do the following:
+
+1. `console sync:data product_concrete_product_offers` exports data from `spy_product_concrete_product_offers_storage` table to Redis.
+2. `console sync:data product_offer` exports data from `spy_product_offer_storage` table to Redis.
+
+Make sure that when the following entities get updated via the ORM, the corresponding Redis keys have the correct values.
```

Comment:
Use "through" instead of "via"
```suggestion
Make sure that when the following entities get updated through the ORM, the corresponding Redis keys have the correct values.
```